ksqldb-engine/src/main/java/io/confluent/ksql/query/KafkaStreamsQueryValidator.java

```diff
@@ -69,10 +77,41 @@ private void validateCacheBytesUsage(
       return;
     }
     final long configured = getCacheMaxBytesBuffering(config);
-    final long usedByRunning = running.stream()
-        .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
-                .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
-        .sum();
+    long usedByRunning;
+    if (!config.getConfig(true).getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+      usedByRunning = running.stream()
+          .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
+              .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
+          .sum();
+    } else {
+      usedByRunning = running.stream()
+          .filter(t -> !(t instanceof BinPackedPersistentQueryMetadataImpl))
+          .mapToLong(r -> new StreamsConfig(r.getStreamsProperties())
+              .getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG))
+          .sum();
+      final Set<String> runtimes = new HashSet<>();
+      Long maxCache = -1L;
```

Comment:
Why does this need to be a `Long` rather than a `long` / why all the casting? It kind of makes it look like there's something much more complicated going on when it's actually quite straightforward logic. Makes me feel like I'm missing something... 🤔

Reply:
also, super nit: `maxCache` --> `cacheSizeBytesPerRuntime` or similar

Reply:
That Long stuff because I was using a map and he casting because the config map is just an object